tests/com.espressif.idf.ui.test/src/com/espressif/idf/ui/test/executable/cases/project/NewEspressifIDFProjectSBOMTest.java

```diff
@@ -0,0 +1,264 @@
+			bot.sleep(1000);
+		}
+
+		private static void givenNewEspressifIDFProjectIsSelected(String category, String subCategory)
+		{
+			Fixture.category = category;
+			Fixture.subCategory = subCategory;
+		}
+
+		private static void givenProjectNameIs(String projectName)
+		{
+			Fixture.projectName = projectName;
+		}
+
+		private static void whenNewProjectIsSelected() throws Exception
+		{
+			ProjectTestOperations.setupProject(projectName, category, subCategory, bot);
+		}
+
+		private static void whenProjectIsBuiltUsingContextMenu() throws IOException
+		{
+			ProjectTestOperations.buildProjectUsingContextMenu(projectName, bot);
+			ProjectTestOperations.waitForProjectBuild(bot);
+			TestWidgetWaitUtility.waitForOperationsInProgressToFinishAsync(bot);
+		}
+
+		private static void cleanTestEnv()
+		{
+			TestWidgetWaitUtility.waitForOperationsInProgressToFinishAsync(bot);
+			ProjectTestOperations.closeAllProjects(bot);
+			ProjectTestOperations.deleteAllProjects(bot);
+		}
+
+		private static void whenOpenSbomTool() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "SBOM Tool");
+		}
+
+		private static void thenRunOKbuttonDisabled(SWTWorkbenchBot bot) throws IOException
+		{
+			SWTBotButton okButton = bot.button("OK");
+			assertTrue(!okButton.isEnabled());
+		}
+
+		private static void whenRedirectOutputToTheFileClicked() throws IOException
+		{
+			bot.checkBox("Redirect output to the file").click();
+		}
+
+		private static void whenRunSbomTool() throws IOException
+		{
+			bot.button("OK").click();
+			ProjectTestOperations
+					.joinJobByName(com.espressif.idf.ui.update.Messages.SbomCommandDialog_EspIdfSbomJobName);
+		}
+
+		private static void thenCheckResultInConsole() throws IOException
+		{
+			ProjectTestOperations.findInConsole(bot, "Espressif IDF Tools Console", "SPDXVersion");
+		}
+
+		private static void thenCheckInConsole() throws IOException
+		{
+			ProjectTestOperations.findInConsole(bot, "Espressif IDF Tools Console",
+					"The output was redirected to the file:");
+		}
+
+		private static void whenRefreshProject() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "Refresh");
+		}
+
+		private static void thenOpenSbomFile() throws IOException
+		{
+			bot.tree().getTreeItem(projectName).getNode("sbom.txt").select();
+			bot.tree().getTreeItem(projectName).getNode("sbom.txt").doubleClick();
+		}
+
+		private static void thenCheckSbomFile() throws IOException
+		{
+			assertTrue(ProjectTestOperations.checkTextEditorContentForPhrase("SPDXVersion", bot));
+		}
+
+		private static void thenAddSpaceToOutputFilePathAndCheckValidation(SWTWorkbenchBot bot) throws IOException
+		{
+			SWTBotButton okButton = bot.button("OK");
+			if (!SystemUtils.IS_OS_LINUX)
+			{
+				bot.shell("Software Bill of Materials Tool").bot().textWithLabel("Output File Path:").setText(" ");
+				assertTrue(!okButton.isEnabled());
+			}
+		}
```

Comment:
this method can be split further into whenSpaceInOutputFilePath .... and the other method thenCheckOkButton (you already have that method)